includes/class-wcs-object-data-cache-manager.php

```diff
@@ -65,6 +65,14 @@ public function __construct( $object_type, $data_keys ) {
 	public function init() {
 		add_action( "woocommerce_before_{$this->object_type}_object_save", [ $this, 'prepare_object_changes' ] );
 		add_action( "woocommerce_after_{$this->object_type}_object_save", [ $this, 'action_object_cache_changes' ] );
+
+		add_action( "woocommerce_before_delete_{$this->object_type}", [ $this, 'prepare_object_to_be_deleted' ], 10, 2 );
+		add_action( "woocommerce_{$this->object_type}_deleted", [ $this, 'deleted' ] );
```

Comment:
```suggestion
		add_action( "{$this->object_type}", [ $this, 'deleted' ] );
```

`woocommerce_subscription_deleted` exists but the order equivalent doesn't. The consistant hook between the two of them is `woocommerce_delete_subscription` and `woocommerce_delete_order`.

Reply:
Thank you for pointing that out. I have address this feedback.